client/lib/reader-post-flux-adapter/index.js

```diff
@@ -42,6 +42,16 @@ const fluxPostAdapter = Component => {
 		state = this.getStateFromStores( this.props );
 
 		updateState = ( newState = this.getStateFromStores() ) => {
+			// check to see if this new state is the same as the old state
+			if ( newState.posts.length === this.state.posts.length ) {
```

Comment:
> Sure, in the end it makes no difference.

No, it does make a difference. Not calling setState when you don't need to means you don't schedule a reconciliation run _at all_. No work is enqueued, nothing further happens. If we defer this work by constantly calling setState when nothing changed (remember, there are more than one of these on a page at a time), we make react do more work than it really needs to do. This shows up in profiling (or at least it did last time I checked) and is the whole reason smart-set-state exists.